pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java

```diff
@@ -117,9 +115,31 @@ void processArguments() throws Exception {}
      * Function level command
      */
     @Getter
-    abstract class FunctionCommand extends NamespaceCommand {
-        @Parameter(names = "--name", description = "The function's name", required = true)
+    abstract class FunctionCommand extends BaseCommand {
+        @Parameter(names = "--fqfn", description = "The Fully Qualified Function Name (FQFN) for the function")
+        protected String fqfn;
+
+        @Parameter(names = "--tenant", description = "The function's tenant")
+        protected String tenant;
+
+        @Parameter(names = "--namespace", description = "The function's namespace")
+        protected String namespace;
+
+        @Parameter(names = "--name", description = "The function's name")
         protected String functionName;
+
+        @Override
+        void processArguments() throws Exception {
+            if (null != fqfn) {
+                String[] fqfnArray = fqfn.split("/");
+                if (fqfnArray.length != 3) {
+                    throw new IllegalArgumentException("Fully qualified function names (FQFNs) must be of the form tenant/namespace/name");
+                }
+                tenant = fqfnArray[0];
```

Comment:
better to throw exception if tenant/namespace/functions are inconsistent.

e.g. `-tenant t1 -fqfn t2/namespace/name`

Reply:
Well, in this case you can *either* set a FQFN *or* tenant + namespace + name. So I think that I'll change this to make sure users don't attempt to apply both systems.